q2_feature_table/tests/test_summarize.py

```diff
@@ -516,5 +490,77 @@ def test_vega_spec_nandling(self):
         self.assertEqual(spec['data'][0]['values'], exp)
 
 
+class TabulateSampleFrequencyTests(TestCase):
+
+    def test_basic_case(self):
+        table = biom.Table(np.array([[25, 25, 25], [25, 25, 25]]),
+                           ['O1', 'O2'],
+                           ['S1', 'S2', 'S3'])
+        obs = tabulate_sample_frequencies(table).to_dataframe()
+
+        exp = pd.DataFrame({'Frequency': ['50.0', '50.0', '50.0'],
+                            'No. of Associated Features':
+                            ['2', '2', '2']},
+                           index=['S1', 'S2', 'S3'])
+        exp.index.name = 'Sample ID'
+        pd.testing.assert_frame_equal(exp, obs)
+
+
+class TabulateFeatureFrequencyTests(TestCase):
+
+    def test_basic_case(self):
+        table = biom.Table(np.array([[25, 25, 25], [25, 25, 25]]),
```

Comment:
Same request as above - make the features not all identical, in terms of both pieces of information you're returning about them.

Reply:
Also, does this handle empty tables? What about empty samples or empty features in those tables? Add tests for these edge cases to all tests.

Reply:
@Oddant1 are there examples of what should happen in these edge cases?

Reply:
@Clockwork-Rat it doesn't looke like we really cover these edge cases for summarize currently. [This](https://github.com/Clockwork-Rat/q2-feature-table/blob/174761b67f9522a17c5e79d3c2e5addd4f2a9e83/q2_feature_table/tests/test_summarize.py#L478) looks like the closest thing we currently have.

If you have empty samples or empty features you're probably expecting. NaNs or Nones in the relevant rows of the interactive sample detail and the feature detail tabs respectively. If the entire thing is empty you should probably just error out (I expect something would cause an error anyway as it stands but idk).

I would try putting your edge case data through the existing summarize viz and see what the results look like/what errors you get honestly. It doesn't look like there are any tests for that, so you can't tell exactly what you're expecting there by reading the tests.